lib/galaxy/datatypes/text.py

```diff
@@ -539,7 +540,7 @@ def set_meta(self, dataset: DatasetProtocol, overwrite: bool = True, **kwd) -> N
         if dataset.has_data():
             with open(dataset.get_file_name()) as fh:
                 try:
-                    json_dict = json.load(fh)
+                    json_dict = json_stream.load(fh)
```

Comment:
We can't tell if this works or not because of the try/except here. I'd suspect though that you have to iterate over this thing and get taxonId and speciesName as you encounter them. Also this is probably a list and not a dict as the name implies, otherwise entry in `for entry in json_dict:` would be the dictionary key ...